### ISSUE TITLE
Determine Asset's Metadata

### ISSUE THREAD

**Issue by Kiarii** [akvo/akvo-lumen#778]
# Context
Various assets have various metadata e.g. An asset has date (created, modified), owner, etc

Currently we have left empty spaces for places designated for metadata e.g.
- In the library/collection
<img width="578" alt="screen shot 2017-05-29 at 18 21 39" src="https://cloud.githubusercontent.com/assets/12692444/26554981/b23d28ba-449b-11e7-97ed-c901d6dd00f3.png">

- Also in asset viewing/edit screen
<img width="310" alt="screen shot 2017-05-29 at 18 22 09" src="https://cloud.githubusercontent.com/assets/12692444/26554994/c046317c-449b-11e7-9976-5c53895f0467.png">

# Proposal
We need to determine what metadata snippet will be meaningful for each asset and in which situation, and if they are need

**Comment by loicsans** [comment]
Let's add the author, while removing the sorting function and keeping last modified only.
Make visualisations icon specific to its type.

**Comment by loicsans** [comment]
Priority: 10
Effort: 4 or 7 Depends on implementation

**Comment by nadiagorchakova** [comment]
If the resource is a dataset, would be really helpful to also have data source in the metadata snippet. Possible data sources at the moment are: Data File, Link, Akvo Flow, GeoTIFF (In addition to what Loic suggested)

**Comment by finnfiddle** [comment]
so just to be clear I need to add `author`, `last modified` and dataset `data source`? @nadiagorchakova @loicsans

**Comment by loicsans** [comment]
At least `author` and `last modified`. Not sure about the datasource. @nadiagorchakova what do you think?

**Comment by nadiagorchakova** [comment]
So we've recently added 'data last updated' date to all visualisations. Do you think it will still add value to have 'last modified' in the library too? I'm fine with not displaying a 'data source' in meta data if it'll make the interface look clunky and unreadable. We could find a better place for 'data source' later. But for sure let's add the 'author'!

<img width="369" alt="screen shot 2018-05-17 at 14 54 23" src="https://user-images.githubusercontent.com/21127166/40178709-917a1748-59e2-11e8-9fcd-842ac86bd4b1.png">

**Comment by finnfiddle** [comment]
## Backend tasks:

- [ ] add author to dashboards (create and retrieve)
- [ ] add author to rasters
- [ ] add author to datasets
- [ ] add dataset source

**Comment by janagombitova** [comment]
@finnfiddle and @tangrammer and @kardan I checked the dark-test environment and noticed that the author is only shown in the library if the user has her first and last name defined. After talking with @nadiagorchakova about it, she said that users very rarely have these two properties defined, email is one each user has. 

<img width="1262" alt="screen shot 2018-07-05 at 09 25 02" src="https://user-images.githubusercontent.com/12456965/42309202-33fbb8d4-8038-11e8-88b3-9ecb2d05836f.png">

We should show the email if the user's name is not defined.

**Comment by finnfiddle** [comment]
@janagombitova it should show either first or last names if only one is set. I cant work out how to test this though because I cant save my profile without both names set. I think what you might be experiencing is the fact that some existing entities will not have an author set because we have added this functionality in this task. so it will just hide it for them. if you add a new entity it should have an author. about the dates - sounds reasonable

**Comment by janagombitova** [comment]
@finnfiddle I see, I will test it out again with just my email address set and when creating a new thing and let you know.

**Comment by janagombitova** [comment]
@finnfiddle we just tested with @nadiagorchakova. I created a new visualisation on dark-test with name and family name defined on my user and it shows properly. She created a new visualisation with her user where only email is set up. For this one nothing shows. 

We should show the user email, if there is no first or/and last name defined for the user. 

<img width="1207" alt="screen shot 2018-07-05 at 12 29 58" src="https://user-images.githubusercontent.com/12456965/42318385-234f7978-804f-11e8-9622-10216615376a.png">
